module4/homework/appendNewRecipe/AppendNewRecipe.cpp

```diff
@@ -0,0 +1,70 @@
+#include <algorithm>
+#include <fstream>
+#include <iterator>
+
+#include <iostream>
+
+#include "AppendNewRecipe.hpp"
+
+bool AppendNewRecipe(std::vector<std::string> steps,
+                     const std::list<std::string>& ingredients,
+                     const std::deque<std::pair<size_t, char>>& amount) {
+    std::fstream recipes("recipes.txt", recipes.out | recipes.app);
+    if (recipes.is_open()) {
+        std::cout << FormatRecipit(steps, ingredients, amount).str();
+        recipes << FormatRecipit(steps, ingredients, amount).str();
+        recipes.close();
+        return true;
+    }
+
+    return false;
+}
+
+std::vector<std::string> FormatIngredients(const std::list<std::string>& ingredients,
+                                           const std::deque<std::pair<size_t, char>>& amount) {
+    std::vector<std::string> result;
+    std::transform(ingredients.begin(), ingredients.end(), amount.begin(), std::back_inserter(result),
```

Comment:
i suggest you to use const iterators whenever you can  ;p

Reply:
done